### PR TITLE
[Placement Group]Fix CommitResources  crash bug

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
image

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc [hunk 1]
```diff
@@ -386,17 +386,23 @@ void GcsPlacementGroupScheduler::CommitAllBundles(
...
+                                      schedule_failure_handler,
+                                      schedule_success_handler](const Status &status) {
+      lease_status_tracker->MarkCommitRequestReturned(node_id, bundle, status);
+      if (lease_status_tracker->AllCommitRequestReturned()) {
+        OnAllBundleCommitRequestReturned(lease_status_tracker, schedule_failure_handler,
+                                         schedule_success_handler);
+      }
+    };
+
+    if (node != nullptr) {
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc:399]: Nice! This is what I needed to do handle TODO there :)

**Review by rkooo567 — APPROVED** [review]
LGTM!

**Review by rkooo567 — APPROVED** [review]

File: src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc [hunk 2]
```diff
@@ -386,17 +386,23 @@ void GcsPlacementGroupScheduler::CommitAllBundles(
...
+      CommitResources(bundle, node, commit_resources_callback);
+    } else {
+      RAY_LOG(WARNING)
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc:402]: I think INFO is fine?

**Review by rkooo567 — APPROVED** [review]

**Comment by ericl** [comment]
Seems like compilation is failing on MAC.

**Comment by ffbin** [comment]
> Seems like compilation is failing on MAC.

CI environment problem, I try it again and pass it.

**Pull request merged**